pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3443,3 +3449,31 @@ func recordAwsMetric(actionName string, timeTaken float64, err error) {
 	}
 
 }
+
+// GetAdditionalTagsFromAnnotation converts the comma separated list of key-value
+// pairs in the "AdditionalTags" annotation and returns it as a map.
+func GetAdditionalTagsFromAnnotation(annotations map[string]string, annotationName string) map[string]string {
+	additionalTags := make(map[string]string)
+	if additionalTagsList, ok := annotations[annotationName]; ok {
+		additionalTagsList = strings.TrimSpace(additionalTagsList)
+
+		// Break up list of "Key1=Val,Key2=Val2"
+		tagList := strings.Split(additionalTagsList, ",")
+
+		// Break up "Key=Val"
+		for _, tagSet := range tagList {
+			tag := strings.Split(strings.TrimSpace(tagSet), "=")
+
+			// Accept "Key=val" or "Key=" or just "Key"
+			if len(tag) >= 2 && len(tag[0]) != 0 {
+				// There is a key and a value, so save it
+				additionalTags[tag[0]] = tag[1]
+			} else if len(tag) == 1 && len(tag[0]) != 0 {
+				// Just "Key"
+				additionalTags[tag[0]] = ""
+			}
+		}
+	}
+
+	return additionalTags
```

Comment:
This should be covered by TestGetAdditionalTagsFromAnnotation in pkg/cloudprovider/providers/aws/aws_test.go (this test already existed, I just slightly adapted it to match the other modifications). Do you have some additional tests in mind? I can of course add them.